refnx/reflect/_mdsimulation.py

```diff
@@ -0,0 +1,307 @@
+    """
+    Determines the scattering length density profile along the z-axis of a
+    molecular dynamics output file (PDB format, unless MDAnalysis is
+    installed). Lateral averaging occurs in the x-y plane.
+
+    Parameters
+    ----------
+    pdbfile: str
+        The path and name of the .pdb file or for which the reflectometry
+        should be found.
+    layer_thickness: float, optional
+        The thickness of the layers that the simulation cell should be sliced
+        into. This will depend on the size of the particles in the simulation,
+        for example a coarse-grained forcefield simulation will require thicker
+        layers to stop overbinning.
+    cut_off: float, optional
+        The thickness (in the z-dimension) of the volume of the bottom
+        simulation cell that should be ignored. This is to allow for the use of
+        a vacuum gap in the simulation cell. This thickness can be determined
+        by visualising the simulation or by running the box with a cut-off of 0
+        and investigating the SLD profile that is produced.
+    flip: bool, optional
+        False if the system should be read as is, True if the simulation cell
+        should be rotated through the xy-plane. This expects the neutrons or
+        X-ray to be incident at the bottom of the simulation cell (where
+        z=0).
+    roughness: float, optional
+        The fractional (of the layer thickness) roughness to be considered
+        between the layers in the simulation cell.
+    verbose: bool, optional
+        True if you want to be notified when the pdb and lgt files have been
+        read and when the sld profile is calculated.
+    structure: biopython.structure, attribute
+        The is the object which contains the information read in from the
+        simulation trajectory.
+    dimensions: float, array_like, attribute
+        An array of 3 floats containing the simulation cell dimensions.
+    """
+    def __init__(self, pdbfile, layer_thickness=1, cut_off=5, flip=False,
+                 roughness=0, verbose=False):
+        self.pdbfile = pdbfile
+        self.structure = None
+        self.dimensions = [0, 0, 0]
+        self.read_pdb()
+        self.verbose = verbose
+        if self.verbose:
+            print('PDB file read.')
+        self.av_layers = np.zeros((int(np.floor(self.dimensions[2] /
+                                                layer_thickness)) + 1, 5))
+        self.av_layers[:, 0] = layer_thickness
+        self.av_layers[:, 3] = layer_thickness * roughness
+        self.layers = np.array([self.av_layers, ] * len(self.structure))
+        self.flip = flip
+        self.cut_off = cut_off
+
+    def assign_scattering_lengths(self, lgtfile=None, radiation='neutron',
+                                  xray_energy=None, atom_types=None,
+                                  scattering_lengths=None):
+        """
+        Assign the scattering lengths either from a .lgt file, a user defined
```

Comment:
It's not explained anywhere what an lgt file is.

Reply:
(although it is below). Perhaps put a description in a `Notes` section.

Reply:
Because it's not clear why the lgt file is required from the docstring, i.e. the coarse grained model.